src/test/java/core/basesyntax/strategy/PurchaseOperationHandlerTest.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.strategy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import org.junit.Before;
+import org.junit.Test;
+
+public class PurchaseOperationHandlerTest {
+    private PurchaseOperationHandler purchaseOperationHandler = new PurchaseOperationHandler();
+    private Fruit fruit;
+
+    @Before
+    public void beforeClass() throws Exception {
+        fruit = new Fruit("apple");
+    }
+
+    @Test
+    public void applyPurchase_OK() {
+        Fruit apple = new Fruit("apple");
+        Storage.storage.put(apple, 14);
+        purchaseOperationHandler.apply(new Transaction("p", new Fruit("apple"), 9));
+        Integer expected = 5;
+        Integer actual = Storage.storage.get(fruit);
+        assertEquals(expected, actual);
+    }
```

Comment:
```suggestion
    public void applyPurchase_OK() {
        Storage.storage.put(fruit, 14);
        purchaseOperationHandler.apply(new Transaction("p", new Fruit("apple"), 9));
        Integer expected = 5;
        Integer actual = Storage.storage.get(fruit);
        assertEquals(expected, actual);
    }
```

 i think change to this and will work , not sure, but i think you have creating more fruits then need... and same with other methods, i don't think you need create each time new object